From 53bfd571a1a01619518cd7925a8c690d239bebc9 Mon Sep 17 00:00:00 2001
From: Vladimir Sokolovsky <vlad@mellanox.com>
Date: Wed, 5 Dec 2012 18:07:03 +0200
Subject: [PATCH 20/20] BACKPORT: Fixed core/netlink compilation on 3.0

Signed-off-by: Vladimir Sokolovsky <vlad@mellanox.com>
---
 drivers/infiniband/core/netlink.c |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/drivers/infiniband/core/netlink.c b/drivers/infiniband/core/netlink.c
index f8d5b0b..ad5d940 100644
--- a/drivers/infiniband/core/netlink.c
+++ b/drivers/infiniband/core/netlink.c
@@ -163,10 +163,14 @@ static int ibnl_rcv_msg(struct sk_buff *skb, struct nlmsghdr *nlh)
 #endif
 				};
 				return netlink_dump_start(nls, skb, nlh, &c);
-#else
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(3,1,0)
 				return netlink_dump_start(nls, skb, nlh,
 							  client->cb_table[op].dump,
 							  NULL, 0);
+#else
+				return netlink_dump_start(nls, skb, nlh,
+							  client->cb_table[op].dump,
+							  NULL);
 #endif /* (LINUX_VERSION_CODE >= KERNEL_VERSION(3,4,0)) */
 			}
 		}
-- 
1.7.1

